prime-router/docs/design/validation/design.md

```diff
@@ -0,0 +1,286 @@
+// load many StructureDefinitions, ValueSets, CodeSystems, etc.
+PrePopulatedValidationSupport prePopulatedSupport = new PrePopulatedValidationSupport(ctx);
+prePopulatedSupport.addStructureDefinition(someStructureDefnition);
+prePopulatedSupport.addValueSet(someValueSet);
+
+// Add the custom definitions to the chain
+supportChain.addValidationSupport(prePopulatedSupport);
+
+// Wrap the chain in a cache to improve performance
+CachingValidationSupport cache = new CachingValidationSupport(supportChain);
+
+// Create a validator using the FhirInstanceValidator module. We can use this
+// validator to perform validation
+FhirInstanceValidator validatorModule = new FhirInstanceValidator(cache);
+FhirValidator validator = ctx.newValidator().registerValidatorModule(validatorModule);
+ValidationResult result = validator.validateWithResult(input);
+```
+</details>
+
+##### Lots more information [here](https://hapifhir.io/hapi-fhir/docs/validation/introduction.html)
+
+##### FHIR Validator Wrapper
+This external project is maintained by the HL7 FHIR Foundation and is an easy to use web tool for uploading data
+for validation against a user-selectable FHIR spec. Implementation guides can be supplied or picked from a pre-
+populated list providing a powerful starting point for our validation tool.
+
+Check out the tool online [here](https://validator.fhir.org). And on [github](https://github.com/hapifhir/org.hl7.fhir.validator-wrapper)!
+This tool can be deployed as is and meet all requirements for our validation tool. Ideally we will provide saved
+configurations equivalent to any validation profiles we have internal to ReportStream to better serve our users' needs.
+
+This tool also serves as amazing sample code for initializing the various classes and interfaces needed to run a 
+validation. To include loading an implementation guide from a URL or using a specific FHIR version. Also of significant
+value is the pre-populated list of implementation guides from various health facilities.
+
+#### Resulting Tickets
+
+- Implement reusable validation component [#8974](https://github.com/CDCgov/prime-reportstream/issues/8974)
+- Research validation configurations: [#8976](https://github.com/CDCgov/prime-reportstream/issues/8976)
+- Add FHIR Sender Converter Validation: [#8978](https://github.com/CDCgov/prime-reportstream/issues/8978)
+- Add FHIR Internal Validation:  [#8979](https://github.com/CDCgov/prime-reportstream/issues/8979)
+- Enable Parser validation and check result: [#8980](https://github.com/CDCgov/prime-reportstream/issues/8980)
+- Configure FHIR parser, sender, internal, and receiver validation profiles: [#9034](https://github.com/CDCgov/prime-reportstream/issues/9034)
+- Add FHIR Receiver Validation: [#9035](https://github.com/CDCgov/prime-reportstream/issues/9035)
+- Test default validation against existing messages: https://github.com/CDCgov/prime-reportstream/pull/9165
+
+### HL7 Validation
```

Comment:
You need to make a decision which tool (NIST vs HAPI) to use and summarize that here and why you chose that method? This might mean you need to talk to Marcia (and maybe Andrew) and ask them why they chose one over the other. That information should be recorded here.

Reply:
Sorry, I thought the decision was clear since the HAPI library is not usable. I will add a paragraph reiterating that.